lib/queue.rb

```diff
@@ -1,31 +1,64 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @size = 20
+    @front = -1
+    @rear = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if (@front == 0 && @rear == @size - 1) ||
+      (@rear == (@front - 1) % (@size - 1))
+      raise RuntimeError.new("Error: Queue is full")
+    elsif @front == -1 # queue is empty
+      @front = @rear = 0
+      @store[@rear] = element
+    elsif @rear == @size - 1 && @front != 0 # rear needs to wrap around
+      @rear = 0
+      @store[@rear] = element
+    else 
+      @rear += 1
+      @store[@rear] = element
+    end
```

Comment:
This could be simplified a bit.

```suggestion
    else 
      @rear = (@rear + 1) % @store.length
      @store[@rear] = element
    end
```